slasher/cache/validators_cache.go

```diff
@@ -0,0 +1,71 @@
+package cache
+
+import (
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// validatorsCacheSize defines the max number of validators public keys the cache can hold.
+	validatorsCacheSize = 300000
+	// Metrics for the validator cache.
+	validatorsCacheHit = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "validators_cache_hit",
+		Help: "The total number of cache hits on the validators cache.",
+	})
+	validatorsCacheMiss = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "validators_cache_miss",
+		Help: "The total number of cache misses on the validators cache.",
+	})
+)
+
+// ValidatorsCache is used to store the public keys needed for signature verification.
+type ValidatorsCache struct {
+	cache *lru.Cache
+}
+
+// NewValidatorsCache initializes the cache.
+func NewValidatorsCache(size int, onEvicted func(key interface{}, value interface{})) (*ValidatorsCache, error) {
```

Comment:
So the DB isn't involved at all for this? Or do you intend to do it later?

Reply:
as discussed the most common use will be through a authorised nodes with in memory public keys. no db at this time